### ISSUE TITLE
Incorrect diagnostic implicitly declaring library function in C89 mode

### ISSUE THREAD

**Issue by efriedma-quic** [llvm/llvm-project#12168]
|  |  |
| --- | --- |
| Bugzilla Link | [11796](https://llvm.org/bz11796) |
| Version | unspecified |
| OS | All |

## Extended Description 
Testcase (compile with clang -std=c89 -pedantic-errors):
```c
int f() { return strcmp("foo", "bar"); }
```
Output:

```
<stdin>:1:18: error: implicitly declaring C library function 'strcmp' with type 'int (const char *, const char *)' [-Werror]
int f() { return strcmp("foo", "bar"); }
                 ^
<stdin>:1:18: note: please include the header <string.h> or explicitly provide a declaration for 'strcmp'
1 error generated.
```
This code is valid in C89, so we shouldn't give an error.  (Granted, I wouldn't suggest writing code like this.)

**Comment by philnik777** [comment]
I don't really see the problem here. You specifically enable `-Werror` via `-pedantic-errors`, so it seems completely fine to generate an error for this.

**Comment by efriedma-quic** [comment]
The fact that we warn here by default isn't really the issue.  The issue here is that -pedantic-errors isn't supposed to promote all warnings to errors; it's supposed to specifically promote diagnostics which indicate standard-conformance violations to errors, and not promote other warnings.  And this isn't a conformance violation in C89, as far as I know.
